### ISSUE TITLE
Cannot change-folder to mailboxes containing white spaces

### ISSUE THREAD

**Issue by grational** [neomutt/neomutt#2226]
Hi, I have experienced an issue related to my macros using the latest release of neomutt (20200320).

The following macro doesn't work as expected:

```macro index,pager ga "<change-folder>=Mail box<enter>" "open a mailbox with whitespace in the name "```

## Expected Behaviour
Pressing the keystrokes _ga_ has to change the folder to _=Mail box_.

## Actual Behaviour

If the macro is executed the prompt states `box is not a mailbox`.

## Steps to Reproduce
Set the macro: `macro index,pager ga "<change-folder>=Mail box<enter>" "open a mailbox with whitespace in the name "` and press `ga` in the index or pager environment.

One can also reproduce the behavior calling the function directly from the prompt:
```
:exec change-folder
Open mailbox ('?' for list):=Mail <-pressing the space here 
Open mailbox ('?' for list):box    <-has removed all the previous text, so just 'box' remain
box is not a mailbox
```

## How often does this happen?

- Always

## When did it start to happen?

- When I upgraded. It was working fine with the tag 20171215 distributed with Lubuntu 18.04

## NeoMutt Version
neomutt version affected: 20200320

```
NeoMutt 20200320
Copyright (C) 1996-2016 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 4.18.0-15-generic (x86_64)
ncurses: ncurses 6.1.20180127 (compiled with 6.1.20180127)
libidn: 1.33 (compiled with 1.33)
GPGME: 1.10.0
GnuTLS: 3.5.18
libnotmuch: 5.0.0
hcache backends: lmdb
hcache compression: zlib

Configure options: --prefix=/usr/local --ssl --gnutls --gpgme --gss --sasl --autocrypt --notmuch --disable-nls --sqlite --zlib --lmdb

Compilation CFLAGS: -g -O2 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/local/include -DNCURSES_WIDECHAR -isystem /usr/include/mit-krb5

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash

Compile options:
  +autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens
  +getaddrinfo +gnutls +gpgme +gss +hcache -homespool +idn +inotify
  -locales_hack -lua -lz4 +meta -mixmaster -nls +notmuch -openssl +pgp +sasl
  +smime +sqlite +start_color +sun_attachment +typeahead +zlib -zstd
MAILPATH="/var/mail"
PKGDATADIR="/usr/local/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/usr/local/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

* Operating System and its version

_Ubuntu 18.04 LTS - Bionic_

* Were you using multiple copies of NeoMutt at once?
_No_

* Were you using 'screen' or 'tmux'?
_No_

* Is your email local (maildir) or remote (IMAP)?
_Remote (IMAP)_

**Comment by grational** [comment]
Unfortunately, it seems that the patch has not fixed my issue. To test it, I have git fetched from the  master branch where @flatcap has merged the two commits of @gahr, make clean && configure && make.

The new neomutt executable continues to state: "Mail is not a mailbox" when I try to automatically change folder to "+[Gmail]/Sent Mail".

I can reproduce it also manually executing: `exec change-folder` and then inserting a space when the `Open mailbox:` prompt appear.

**Comment by gahr** [comment]
Right - this is an issue with parsing the input line, not with trying to actually parse the path.

**Comment by gahr** [comment]
This is due to O__EDITOR_MAILBOX_CYCLE kicking in in `mutt_enter_string_full`

**Comment by gahr** [comment]
@grational , I and @flatcap have gone through this and we cannot reproduce being able to type in mailboxes with spaces on 20171215.

What happens is that the space is bound to `mailbox-cycle`, which is a function you can use to repeatedly populate the prompt with the next mailbox with unread mail:

1. launch neomutt
2. run `:exec check-stats`
3. `c`
4. start hitting space, you should see the prompt being completed

As far as we can see (and could reproduce), this was also the case with previous versions. FWIW, I was able to reproduce the behaviour I just described with vanilla mutt-1.13.4.

I guess a workaround would be to `:unbind index,editor <space>`.

Can you please double check that your macro actually worked previously, and if so, provide a minimal but complete config file that we can use to try and reproduce the issue?

**Comment by grational** [comment]
Thank you @gahr and @flatcap for your extensive test. As you have said, I have been able to reproduce the issue also on the 20171215 version shipped with my distribution (ubuntu 18.04).

Also, the workaround you have mentioned has remembered me something I have already seen. So, I have recovered from the vim backup files the previous configuration I was using and it was there:
```
bind editor <space> noop
``` 
When I have deployed my neomutt configuration on github I removed that command... since it was uncommented I was thinking it was something related to the e-mail composition.
Now it's really clear why it was there, what I can say, sorry for being a newbie! :)

Anyway, do you think the default behavior it's something sensible?

**Comment by gahr** [comment]
Thanks for double checking @grational ! To your last question, I don't have a strong opinion. I tend to prefer not to break existing configuration and to keep backwards compatibility, though.
In hindsight, the default is probably not too wise, but I don't think mailboxes with spaces were common in the mbox-only era when this was designed.

I'll close this. Feel free to open a discussion issue if you want to pursue a change in the default key binding. Thanks!
